app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,65 @@
+class PasswordResetsController < ApplicationController
+  before_action :get_user, only: [ :edit, :update ]
+  before_action :valid_user, only: [ :edit, :update ]
+  before_action :check_expiration, only: [ :edit, :update ]
+
+  def new
+  end
+
+  def create
+    @user = User.find_by(email: password_reset_params[:email].downcase)
+    if @user
+      @user.create_reset_digest
+      @user.send_password_reset_email
+      flash[:info] = "Email sent with password reset instructions"
+      redirect_to root_url
+    else
+      flash.now[:danger] = "Email address not found"
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def edit
+  end
+
+  def update
+    if user_params[:password].blank?
+      @user.errors.add(:password, "can't be empty")
+      render :edit, status: :unprocessable_entity
+    elsif !@user.update(user_params)
```

Comment:
viết cái logic này trước thì sẽ dễ hiểu hơn k e
```suggestion
    elsif @user.update(user_params)
```